web/src/main/java/org/fao/geonet/monitor/onlineresource/wps/CallableWpsCheck.java

```diff
@@ -0,0 +1,236 @@
+                "      </wps:Data>\n" +
+                "    </wps:Input>\n" +
+                "    <wps:Input>\n" +
+                "      <ows:Identifier>format</ows:Identifier>\n" +
+                "      <wps:Data>\n" +
+                "        <wps:LiteralData>text/csv</wps:LiteralData>\n" +
+                "      </wps:Data>\n" +
+                "    </wps:Input>\n" +
+                "  </wps:DataInputs>\n" +
+                "  <wps:ResponseForm>\n" +
+                "    <wps:ResponseDocument storeExecuteResponse=\"true\" status=\"true\">" +
+                "      <wps:Output asReference=\"true\" mimeType=\"application/x-netcdf\">" +
+                "        <ows:Identifier>result</ows:Identifier>" +
+                "      </wps:Output>" +
+                "    </wps:ResponseDocument>" +
+                "  </wps:ResponseForm>\n" +
+                "</wps:Execute>";
+    }
+
+
+    private CheckResult submitAndWaitToComplete(String url, String requestXml, int timeoutSeconds, int pollIntervalSeconds) {
+
+        try {
+            DateTime start = new DateTime();
+            DateTime now;
+
+            String statusUrl = submitHttpPost(url, requestXml);
+            logger.info("Submitted request to [" + url + "].  Status URL [" + statusUrl + "]");
+
+            System.out.println("Waiting for process to complete...");
+
+            boolean completed  = false;
+            //  Poll the status URL & wait for the document to contain a
+            //  ExceptionReport or an ExecuteResponse
+            while(completed == false) {
+
+                //  Calculate elapsed seconds
+                DateTime loopStart = new DateTime();
+                Seconds elapsedSeconds = Seconds.secondsBetween(start, loopStart);
+                logger.info("Seconds elapsed waiting: " + elapsedSeconds.getSeconds());
+                if(elapsedSeconds.getSeconds() > timeoutSeconds) {
+                    logger.error("Process execution time [" + elapsedSeconds.getSeconds() + "s] has exceeded the timeout [" + timeoutSeconds + "s]");
+                    completed = true;
+                    return new CheckResult(CheckResultEnum.FAIL, "Process execution timed out after [" + elapsedSeconds.getSeconds() + " seconds]");
+                }
+
+
+                //  Read status document and determine whether the process has
+                //  completed.
+                ExecuteResponse response = getExecuteResponse(statusUrl);
+
+                if(response != null) {
+                    if (response.isSetStatus()) {
+                        StatusType responseStatus = response.getStatus();
+                        if (responseStatus.isSetProcessSucceeded()) {
+                            logger.info("  -  Process status : SUCCEEDED");
+                            completed = true;
+                            return new CheckResult(CheckResultEnum.SUCCESS, "Process execution successful.");
```

Comment:
Perhaps check we actually got some data - we should have.

Reply:
That makes sense - will change to request CSV & check that we got at least one row back.

Reply:
Now verifying data was returned.